src/inout.c

```diff
@@ -66,6 +67,15 @@ int H2T_loadNpy(MMG5_pMesh mmgMesh, int** tabhex, char* filename) {
     return -1;
   }
 
+  /* Read data type*/
+  fseek(inm,0,0);
+  while (!(buffer == 0x3c)) {
```

Comment:
Not easy to understand : `0x3c` is the `<` char or the `60` decimal value. I haven't found any documentation on that : is it a specific marker inside the header line or the length of the header minus 3 char (the header being assumed of constant size)?

If this specific char has a sense,  I propose to use a C macro with an explicit variable name to define this hex value and to use the macro inside the code, so the code will be more understandable (`#define H2T_HEX_<EXPLICIT_NAME> 0x3c`).

By reading the numpy doc (https://numpy.org/devdocs/reference/generated/numpy.lib.format.html), it is difficult to understand why you are reading 2 characters after that and then the dataSize....